analysis/active_analyses.R

```diff
@@ -0,0 +1,520 @@
+                                cox_start = cox_start,
+                                cox_stop = cox_stop,
+                                study_start = ifelse(c=="prevax", prevax_start, vax_unvax_start),
+                                study_stop = study_stop,
+                                cut_points = ifelse(c=="prevax", prevax_cuts, vax_unvax_cuts),
+                                controls_per_case = controls_per_case,
+                                total_event_threshold = total_event_threshold,
+                                episode_event_threshold = episode_event_threshold,
+                                covariate_threshold = covariate_threshold,
+                                age_spline = TRUE,
+                                analysis = "sub_smoking_ever")
+            
+            ## analysis: sub_smoking_current -----------------------------------------------
+            df[nrow(df)+1,] <- c(population = p,
+                                cohort = c,
+                                exposure = exposure, 
+                                outcome = i,
+                                ipw = ipw, 
+                                strata = strata,
+                                covariate_sex = covariate_sex,
+                                covariate_age = covariate_age,
+                                covariate_other = gsub("cov_cat_smoking_status;","",all_covars), #-smoking status
+                                cox_start = cox_start,
+                                cox_stop = cox_stop,
+                                study_start = ifelse(c=="prevax", prevax_start, vax_unvax_start),
+                                study_stop = study_stop,
+                                cut_points = ifelse(c=="prevax", prevax_cuts, vax_unvax_cuts),
+                                controls_per_case = controls_per_case,
+                                total_event_threshold = total_event_threshold,
+                                episode_event_threshold = episode_event_threshold,
+                                covariate_threshold = covariate_threshold,
+                                age_spline = TRUE,
+                                analysis = "sub_smoking_current")
+    
+        }
+
+    }
+
+}
+
+df$name <- paste0("cohort_",df$cohort, "-", 
+                  df$analysis, "-", 
+                  gsub("out_date_","",df$outcome), "-", 
+                  df$population)
+
+# Pre-existing populations only run the outcomes Pneumonia and Pulmonary fibrosis
+# Remove Pre-existing & asthma/copd rows
+df <- df[df$population != "preex_TRUE" | !df$outcome %in% c("out_date_asthma", "out_date_copd"),]
+# Remove history of asthma from covariates for the sample with no preexisiting condition
+df$covariate_other <- ifelse(
+  df$population == "preex_FALSE",
+  sapply(
+    strsplit(df$covariate_other, ";"),
+    function(covars) paste0(covars[covars != "cov_bin_history_asthma_snomed"], collapse = ";")
```

Comment:
History of COPD covariate will need to be removed too.

Reply:
In the seperate PR for considering outcomes from primary care, secondary care and deaths, I will also rename these covariates to `cov_bin_history_asthma` and `cov_bin_history_copd` .